streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java

```diff
@@ -315,13 +290,15 @@ public String queryableStoreName() {
         builder.addGraphNode(this.streamsGraphNode, tableNode);
 
         return new KTableImpl<>(
-            builder,
             name,
-            processorSupplier,
+            materialized != null ? materialized.keySerde() : null,
+            materialized != null ? materialized.valueSerde() : null,
```

Comment:
In this case, we can indeed forward `valSerde` from `Materialized` since the materialized object is for the transformed KTable. Again, if it is null that is also fine passing in, as the key/val Serde objects are null-able.

Reply:
that's not what I mean -- I meant the "else" branch that returns `null` -- even if we get a `valueSerde` from upstream, we cannot forward it, but we can only forward if provided via `Materialized`. This is different to what we can do for the key.

Reply:
We actually can since the materialized object may contain key/value serdes for the resulted KTable.

Reply:
Ignore my comment above: I was not seeing the latest comment from you.

Reply:
> from upstream, we cannot forward it, but we can only forward if provided via Materialized. This is different to what we can do for the key.

I see, yes I can add the comment.